pkg/apis/observability/v1alpha1/objectstorage.go

```diff
@@ -0,0 +1,121 @@
+package v1alpha1
+
+type S3Spec struct {
+	// Bucket is the name of the S3 bucket.
+	// +kubebuilder:validation:Required
+	Bucket string `json:"bucket"`
+	// Endpoint is the S3 endpoint URL.
+	// +kubebuilder:validation:Required
+	Endpoint string `json:"endpoint"`
+	// AccessKeyID is the access key ID for the S3 bucket.
+	// +kubebuilder:validation:Required
+	AccessKeyID string `json:"accessKeyID"`
+	// AccessKeySecret is a reference to a secret containing the access key secret for the S3.
+	// +kubebuilder:validation:Required
+	AccessKeySecret SecretKeySelector `json:"accessKeySecret,omitempty"`
+	// Region is the region where the S3 bucket is located.
+	// +kubebuilder:validation:Optional
+	Region string `json:"region,omitempty"`
+}
+
+type S3STSpec struct {
+	// Bucket is the name of the S3 bucket.
+	// +kubebuilder:validation:Required
+	Bucket string `json:"bucket"`
+	// RoleARN is the ARN of the IAM role to assume for accessing the S3 bucket.
+	// +kubebuilder:validation:Required
+	RoleARN string `json:"roleARN"`
+	// Region is the region where the S3 bucket is located.
+	// +kubebuilder:validation:Optional
+	Region string `json:"region,omitempty"`
+}
+
+type S3CCOSpec struct {
```

Comment:
Sorry for my ignorance but what is CCO? Should it be COO as `json:"s3COO,omitempty` would suggest?